### ISSUE TITLE
BackboneFinetuning - even with train_bn batch normalization is still learning.

### ISSUE THREAD

**Issue by perliczka1** [Lightning-AI/lightning#13902]
## 🐛 Bug

When I use BackboneFinetuning callback with `train_bn=False` batch normalization biases and weights are still changing from the very first epoch, even though they should be fixed until `unfreeze_backbone_at_epoch`.

### To Reproduce

Below is the notebook. I marked with comments all the placed in the default code I modified:
https://colab.research.google.com/drive/1LPiOljLEbFXUUST3Hr4oeQfZlmJFUndZ?usp=sharing

### Expected behavior
 Batch normalization biases and weights (but not means and vars) should be fixed at the beginning of the training. I think it can be achieved by passing `self.train_bn` to `self.freeze` in `freeze_before_training` method.
Later we need to unfreeze it by passing in `unfreeze_and_add_param_group` `train_bn=True`.

### Environment

* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           11.3
* Packages:
	- lightning:         None
	- lightning_app:     None
	- numpy:             1.21.6
	- pyTorch_debug:     False
	- pyTorch_version:   1.12.0+cu113
	- pytorch-lightning: 1.6.5
	- tqdm:              4.64.0
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.7.13
	- version:           #1 SMP Sun Apr 24 10:03:06 PDT 2022

**Comment by rohitgr7** [comment]
`train_bn` argument in the constructor is meant for the backbone after it's unfreezed. If you want to freeze the batch-norm layers from the beginning, you can do:
```py
class CustomBackboneFineTuning(BackboneFinetuning):
    def freeze_before_training(self, pl_module):
        self.freeze(pl_module.backbone, train_bn=False)


trainer = Trainer(..., callbacks=[CustomBackboneFineTuning()])
```

**Comment by perliczka1** [comment]
> `train_bn` argument in the constructor is meant for the backbone after it's unfreezed. If you want to freeze the batch-norm layers from the beginning, you can do:
> 
> ```python
> class CustomBackboneFineTuning(BackboneFinetuning):
>     def freeze_before_training(self, pl_module):
>         self.freeze(pl_module.backbone, train_bn=False)
> 
> 
> trainer = Trainer(..., callbacks=[CustomBackboneFineTuning()])
> ```

It's not clearly stated in the documentation here https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.callbacks.BackboneFinetuning.html

train_bn([bool](https://docs.python.org/3/library/functions.html#bool)) – Whether to make Batch Normalization trainable.

So the intended behavior is that first we freeze the whole backbone except its batch normalization layers, then we unfreeze the backbone and we can freeze its batch normalization layers for the whole training if we want?

**Comment by rohitgr7** [comment]
I think, we can update the docstrings.

**Comment by sirtris** [comment]
I think it is a bit confusing that the `freeze` function can take `train_bn` as an argument if it is not intended to be used there. 
So maybe you could also remove it from there when you change the docstrings.

**Comment by bencwallace** [comment]
Rather than modifying the docstrings to describe this unusual behavior, wouldn't it be better to just change the implementation of `FreezeBackbone` so that it freezes all the weights?
